### PR TITLE
this commit will add search bar functionality to the front end

### PR DESCRIPTION
**Pull request by Brianna-Flo**
## Description
<what this pull request is doing>
Implement the searchbar to filter through board list and store matches into state variable array that is lifted up to app component
Based on code feedback, added error state variable for no results, utilized map for boards, created functions in utils file to abstract out internal logic, created category button component to avoid repetition, app.jsx functions only set state
<what will be done in later PRs and not included here>
Styling of the search bar

## Milestones
<the milestones or stories/features that this works towards>
kudos board frontend

## Resources
<links to tutorials, code snippets, inspirations>
inspiration from frontend from given example: https://kudos-board-exemplar-bck7.onrender.com/
<if you took or translated specific code from the internet, please call it out here!>

## Test Plan
<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>
Tested edge case if no search results: present error state
Edge case if search is requested but no boards: returns error state

### CODE REVIEW

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -15,35 +14,57 @@ const App = () => {
...
-        {/* <Toolbar onCreate/> */}
       </header>
       <main>
-        <BoardList boardList={boards} />
-
-        {/* <BoardPage boardInfo={boards[0]} /> */}
-        {/* <CreateCard/> */}
+        {
+          // if in search mode, present search boards, otherwise present list of boards
+          searchMode ? <BoardList boardList={searchedBoards} searchMode={searchMode} noResults={noResults}/> : <BoardList boardList={boards} searchMode={searchMode} noResults={noResults}/>
```
Comment (paulshaoyuqiao) [inline frontend/src/App.jsx]: This could probably be 
`<BoardList boardList={searchMode ? searchedBoards L boards} searchMode={searchMode} noResults={noResults}/>` instead?

File: frontend/src/utils/utils.js [hunk 2]
```diff
@@ -1,3 +1,10 @@
 const categoryOptions = ["All", "Recent", "Celebration", "Thank You", "Inspiration"]
 
-export {categoryOptions}
\ No newline at end of file
+const searchBoards = (boards, searchTerm) => {
```
Comment (paulshaoyuqiao) [inline frontend/src/utils/utils.js:3]: nit: the naming here could be slightly clearer (more action oriented) because you also have some other variables called `searchedBoards` (gets confusing very quickly :) ). 

A better name could be `performSearchOnBoards` or `findBoardsBySearchTerm`
